pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3602,6 +3603,15 @@ public Position getLastPosition() {
         return ledger.getLastConfirmedEntry();
     }
 
+    @Override
+    public CompletableFuture<Position> getLastCanDispatchPosition() {
```

Comment:
Maybe we can make the method private or move the logics to `getLastCanDispatchPositionWithTxn()`, just expose 
`getLastCanDispatchPositionWithTxn()`.

Reply:
About the simple method comment, I know your intention.

After these change, there will be fewer methods. But this logic runs into the `persistentTopic`.
```java
        PositionImpl tnxMaxReadPosition = transactionBuffer.getMaxReadPosition();
        if (getLastPosition() == tnxMaxReadPosition) {
            return topic.getLastCanDispatchPosition();
        } else {
            return CompletableFuture.completedFuture(tnxMaxReadPosition);
        }
```

I still want it to be cohesive in class `TransactionBuffer`.

I think makes sense to expose the `getLastCanDispatchPosition` method in `TransactionBuffer` and `Topic`, it can clearly remind the caller:  **The position returned by the `getLastPosition` method is not dispatchable, should use `getLastCanDispatchPosition`**

Reply:
For TransactionBuffer, it just handle the TXN commit/abort/append requests and provide the guarantee of READ_COMMIT.
Determine the position to dispatch is Topic's duty, TransactionBuffer should not be aware of it.
See also https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L4122-L4124